lib/floe/workflow/runner.rb

```diff
@@ -30,7 +30,36 @@ def for_resource(resource)
         end
       end
 
-      def run!(image, env = {}, secrets = {})
+      def run!(resource, env = {}, secrets = {})
+        ref, secret = run_async!(resource, env, secrets)
+
+        sleep(1) while running?(ref)
```

Comment:
Should there by an upper limit on this?  (Maybe not for this PR, but perhaps we can make that configurable).